gravitee-am-ui/src/app/domain/settings/password-policy/domain-password-policy.component.ts

```diff
@@ -13,158 +13,95 @@
     private dialogService: DialogService,
   ) {}
 
   ngOnInit() {
     this.domain = this.route.snapshot.data['domain'];
     this.domainId = this.domain.id;
-    this.passwordSettings = this.domain.passwordSettings;
-    if (this.passwordSettings != null) {
-      this.minLength = this.passwordSettings.minLength;
-      this.maxLength = this.passwordSettings.maxLength;
-      this.includeNumbers = this.passwordSettings.includeNumbers;
-      this.includeSpecialCharacters = this.passwordSettings.includeSpecialCharacters;
-      this.lettersInMixedCase = this.passwordSettings.lettersInMixedCase;
-      this.maxConsecutiveLetters = this.passwordSettings.maxConsecutiveLetters;
-      this.excludePasswordsInDictionary = this.passwordSettings.excludePasswordsInDictionary;
-      this.excludeUserProfileInfoInPassword = this.passwordSettings.excludeUserProfileInfoInPassword;
-      this.expiryDuration = this.passwordSettings.expiryDuration;
-      this.passwordHistoryEnabled = this.passwordSettings.passwordHistoryEnabled;
-      this.oldPasswords = this.passwordSettings.oldPasswords;
-    }
-    this.editMode = this.authService.hasPermissions(['application_settings_update']);
+    this.passwordPolicy = {};
+    this.editMode = this.authService.hasPermissions(['domain_settings_update']);
   }
 
   formChange(): void {
     this.formChanged = true;
   }
 
   setIncludeNumbers(e) {
-    this.includeNumbers = e.checked;
+    this.passwordPolicy.includeNumbers = e.checked;
   }
 
   setIncludeSpecialCharacters(e) {
-    this.includeSpecialCharacters = e.checked;
+    this.passwordPolicy.includeSpecialCharacters = e.checked;
   }
 
   setLettersInMixedValue(e) {
-    this.lettersInMixedCase = e.checked;
+    this.passwordPolicy.lettersInMixedCase = e.checked;
   }
 
   setExcludePasswordsInDictionary(e) {
-    this.excludePasswordsInDictionary = e.checked;
+    this.passwordPolicy.excludePasswordsInDictionary = e.checked;
   }
 
   setExcludeUserProfileInfoInPassword(e) {
-    this.excludeUserProfileInfoInPassword = e.checked;
+    this.passwordPolicy.excludeUserProfileInfoInPassword = e.checked;
   }
 
   setPasswordHistoryEnabled(e) {
-    this.passwordHistoryEnabled = e.checked;
+    this.passwordPolicy.passwordHistoryEnabled = e.checked;
   }
 
-  update() {
-    if (this.passwordHistoryEnabled && (!this.oldPasswords || this.oldPasswords < 1 || this.oldPasswords > 24)) {
+  save() {
+    if (
+      this.passwordPolicy.passwordHistoryEnabled &&
+      (!this.passwordPolicy.oldPasswords || this.passwordPolicy.oldPasswords < 1 || this.passwordPolicy.oldPasswords > 24)
+    ) {
       this.snackbarService.open('Number of old passwords must be within the range [1, 24]');
       return;
     }
-
-    const data: any = {};
-
-    if (this.minLength && this.minLength <= 0) {
+    if (this.passwordPolicy.minLength && this.passwordPolicy.minLength <= 0) {
       this.snackbarService.open('Min length must be greater than zero');
```

Comment:
suggestion: change string to  Min length must be greater than zero and smaller then Max Length